src/background_jobs.rs

```diff
@@ -0,0 +1,78 @@
+use crate::core::util;
+use crate::storage::constants::PAGE_SIZE_MAX;
+use crate::storage::db::PageOptions;
+use crate::storage::store::stores::Stores;
+use crate::storage::store::BlockStore;
+use std::collections::HashMap;
+use tokio::sync::watch;
+use tokio::time::Duration;
+use tokio_cron_scheduler::{Job, JobSchedulerError};
+use tracing::{error, info};
+
+pub fn job_block_pruning(
+    schedule: &str,
+    block_retention: Duration,
+    block_store: BlockStore,
+    shard_stores: HashMap<u32, Stores>,
+    sync_complete_rx: watch::Receiver<bool>,
+) -> Result<Job, JobSchedulerError> {
+    Job::new(schedule, move |_, _| {
+        // Wait for sync to complete before pruning
+        let sync_complete = *sync_complete_rx.borrow();
+        if !sync_complete {
+            info!("Sync not complete, skipping block pruning");
+            return;
+        }
+
+        let cutoff_timestamp =
+            util::get_farcaster_time().unwrap() - (block_retention.as_secs() as u64);
+        let stop_height = block_store
+            .get_next_height_by_timestamp(0, cutoff_timestamp)
```

Comment:
We shouldn't require the shard id as a param here, since each store should know the shard if to use. What does it mean if the caller provides a shard id that doesn't match the store?

The shard stores don't hold the shard id, but let's provide it in the constructor so they can use it automatically.